diff --git a/NAM/convnet.cpp b/NAM/convnet.cpp
index 0a9e275..bdf01a8 100644
--- a/NAM/convnet.cpp
+++ b/NAM/convnet.cpp
@@ -355,7 +355,9 @@ std::unique_ptr<nam::ModelConfig> nam::convnet::create_config(const nlohmann::js
   return c;
 }

+#ifndef NAM_DISABLE_JSON_REGISTRY
 namespace
 {
 static nam::ConfigParserHelper _register_ConvNet("ConvNet", nam::convnet::create_config);
 }
+#endif
diff --git a/NAM/dsp.cpp b/NAM/dsp.cpp
index 2cabf22..9366af8 100644
--- a/NAM/dsp.cpp
+++ b/NAM/dsp.cpp
@@ -330,10 +330,12 @@ std::unique_ptr<nam::ModelConfig> nam::linear::create_config(const nlohmann::jso
 }

 // Register the config parser
+#ifndef NAM_DISABLE_JSON_REGISTRY
 namespace
 {
 static nam::ConfigParserHelper _register_Linear("Linear", nam::linear::create_config);
 }
+#endif

 // NN modules =================================================================

diff --git a/NAM/lstm.cpp b/NAM/lstm.cpp
index 2828d50..7d29726 100644
--- a/NAM/lstm.cpp
+++ b/NAM/lstm.cpp
@@ -194,7 +194,9 @@ std::unique_ptr<nam::ModelConfig> nam::lstm::create_config(const nlohmann::json&
 }

 // Register the config parser
+#ifndef NAM_DISABLE_JSON_REGISTRY
 namespace
 {
 static nam::ConfigParserHelper _register_LSTM("LSTM", nam::lstm::create_config);
 }
+#endif
diff --git a/NAM/wavenet.cpp b/NAM/wavenet.cpp
index 4f7e179..4fd1623 100644
--- a/NAM/wavenet.cpp
+++ b/NAM/wavenet.cpp
@@ -919,7 +919,9 @@ std::unique_ptr<nam::ModelConfig> nam::wavenet::create_config(const nlohmann::js
 }

 // Register the config parser
+#ifndef NAM_DISABLE_JSON_REGISTRY
 namespace
 {
 static nam::ConfigParserHelper _register_WaveNet("WaveNet", nam::wavenet::create_config);
 }
+#endif
